dulwich/porcelain.py

```diff
@@ -488,7 +488,7 @@ def init(path=".", *, bare=False, symlinks: Optional[bool] = None):
 
 def encode_refspecs(refspecs, refspec_encoding):
     if refspecs is None:
-        return [b"HEAD"]
+        return [b"HEAD", b"refs/"]
```

Comment:
This is significantly changing the behaviour, as it means not just discovering all refs but also fetching them and populating the local repository with them.